src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const firstCopy = chatMessages.map((chat) => {
+    return {...chat};
+    } );
+
+  const [entries, setEntries] = useState(firstCopy);
+
+  const updateHeartLikes = (chatId) => {
+    const newChatList =[];
+    for (const chat of entries) {
+      if (chat.id !== chatId) {
+        newChatList.push(chat);
+      } else {
+        const newChat ={
+          ...chat,
+          liked: !chat.liked,
+        };
+        newChatList.push(newChat);
+      }
+      setEntries(newChatList)
+    }
+}; 
+    const heartLikes = () =>{
+      return entries.reduce((accumulator, count) => {
+        return count.liked ? accumulator +1 : accumulator;
+    }, 0 );
+  };
```

Comment:
Instead of counting all of the hearts every time, it would be a good idea to keep track of the number of hearts in state, like you have set up in the Post component. Moving that code from Post to the App component would be good. Counting all of the hearts is O(n) time. Using state to keep track of the count is O(1) time.